File: XOR_Cipher/xorCipher.js
function crypt(input, key) {
  let result = '';
  for (let i=0;i<input.length;i++) {
    result += String.fromCharCode(input.charCodeAt(i) ^ key.charCodeAt(i%key.length));
  }
  return result;
}

const key = 'RandomString'
const encryptedString = crypt('this is a test string 123456', key);
console.log('Encrypted: ', encryptedString);
const decryptedString = crypt(encryptedString, key);
console.log('Decrypted: ', decryptedString);

File: XOR_Cipher/xorCipher.cpp
#include<bits/stdc++.h>
using namespace std;

string crypt(string input, string key) {
  int inputLength = input.length();
  int keyLength = key.length();

  string result = "";
  for(int i = 0; i < inputLength; i++) {
    result += input[i] ^ key[i%keyLength];
  }

  return result;
}

int main(){
  string key = "RandomString";
  string testString = "this is a test string 123456";
  string encryptedString = crypt(testString, key);
  cout << "Encrypted: " << encryptedString << endl;

  string decryptedString = crypt(encryptedString, key);
  cout << "Decrypted: " << decryptedString << endl;
}
